src/selectors/move.ts

```diff
@@ -0,0 +1,113 @@
+import { getThoughtById, getChildrenRankedById } from '.'
+import { Index, Lexeme, Parent, Path, PendingMerge, State } from '../@types'
+import { normalizeThought, appendToPath, pathToContext, timestamp, head } from '../util'
+import { getSessionId } from '../util/sessionManager'
+import mergeThoughts from './mergeThoughts'
+
+interface RecursiveMoveData {
+  thoughtIndexUpdates: Index<Lexeme | null>
+  contextIndexUpdates: Index<Parent | null>
+  pendingMerges: PendingMerge[]
+}
+
+/**
+ * Moves a thought to a destination and handles duplicate merging if found any.
+ */
+const move = (
```

Comment:
Why is this a selector rather than a reducer? Selectors are meant to select i.e. read data, so this is a bit confusing.